spec/unit/coinbase/smart_contract_spec.rb

```diff
@@ -198,6 +198,1229 @@
+                                                  'value' => bytes_value
+                                                })
+          )
+        end
+
+        it 'returns the parsed fixed bytes value' do
+          expect(result).to eq(bytes_value)
+        end
+      end
+
+      describe 'bytes24' do
+        let(:method_name_for_context) { 'pureBytes24' }
+        let(:bytes_value) { '0x0102030405060708090a0b0c0d0e0f101112131415161718' }
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'bytes24',
+                                                  'value' => bytes_value
+                                                })
+          )
+        end
+
+        it 'returns the parsed fixed bytes value' do
+          expect(result).to eq(bytes_value)
+        end
+      end
+
+      describe 'bytes25' do
+        let(:method_name_for_context) { 'pureBytes25' }
+        let(:bytes_value) { '0x0102030405060708090a0b0c0d0e0f10111213141516171819' }
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'bytes25',
+                                                  'value' => bytes_value
+                                                })
+          )
+        end
+
+        it 'returns the parsed fixed bytes value' do
+          expect(result).to eq(bytes_value)
+        end
+      end
+
+      describe 'bytes26' do
+        let(:method_name_for_context) { 'pureBytes26' }
+        let(:bytes_value) { '0x0102030405060708090a0b0c0d0e0f101112131415161718191a' }
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'bytes26',
+                                                  'value' => bytes_value
+                                                })
+          )
+        end
+
+        it 'returns the parsed fixed bytes value' do
+          expect(result).to eq(bytes_value)
+        end
+      end
+
+      describe 'bytes27' do
+        let(:method_name_for_context) { 'pureBytes27' }
+        let(:bytes_value) { '0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b' }
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'bytes27',
+                                                  'value' => bytes_value
+                                                })
+          )
+        end
+
+        it 'returns the parsed fixed bytes value' do
+          expect(result).to eq(bytes_value)
+        end
+      end
+
+      describe 'bytes28' do
```

Comment:
We should do this programmatically instead of having an explicit test case for each different bytes amount